site/docs/develop/spec.md

```diff
@@ -139,8 +139,27 @@ recorded within the [_Put Operation_](#put-operation) of a Nessie commit.
 
 ## Operations in a Nessie commit
 
-Each Nessie commit carries one or more operations. Each operation contains the Content Key and
-comes in one of the following variations.
+Each Nessie commit carries one or more operations. Each operation contains the
+[Content Key](#content-key) and is either a [Put](#put-operation), [Delete](#delete-operation) or
+[Unmodified](#unmodified-operation) operation.
+
+One [Content Key](#content-key) must only occur once in a Nessie commit.
```

Comment:
`A [Content Key](#content-key) must only occur at most once in a Nessie commit.`?

Reply:
Hm - that would allow a content-key to occur in multiple operations.

Reply:
Right, e.g. for a table re-create: `Delete(key, oldId)` + `Put(key, newId)`

Reply:
Nope. That would be:
`Delete(oldKey)` +
`Put(newKey, newContentId)`

Reply:
^ that's for a "rename", sorry

A recreate with the same key would manifest as a single `Put` for the current `key`.

Reply:
interesting...  understood :)